spec/requests/audio_recordings/downloader_spec.rb

```diff
@@ -0,0 +1,196 @@
+    end
+
+    it 'contains an empty string in user_name when and anonymous user downloads it' do
+      get '/audio_recordings/downloader', headers: api_request_headers(no_token)
+      expect_success
+
+      expect(response.body).to include('    $user_name = ""')
+    end
+
+    it 'contains the app version and host url when downloaded' do
+      get '/audio_recordings/downloader', headers: api_request_headers(reader_token)
+      expect_success
+
+      expect(response.body).to include("Version #{Settings.version_string} from http://localhost:3000")
+      expect(response.body).to include('  $workbench_url = "http://localhost:3000"')
+    end
+
+    it 'contains a default filter' do
+      get '/audio_recordings/downloader', headers: api_request_headers(reader_token)
+      expect_success
+
+      expect(response.body).to include <<~POWERSHELL
+        $filter = @'
+        {"sorting":{"order_by":"recorded_date","direction":"desc"},"paging":{"items":25},"projection":{"include":["id","recorded_date","site_id","original_file_name"]}}
+        '@
+      POWERSHELL
+    end
+
+    it 'contains a filter set from query string parameters' do
+      get '/audio_recordings/downloader?items=5&filter_partial_match=mp3',
+        headers: api_request_headers(reader_token)
+      expect_success
+
+      expect(response.body).to include <<~POWERSHELL
+        $filter = @'
+        {"filter":{"or":{"media_type":{"contains":"mp3"},"status":{"contains":"mp3"},"original_file_name":{"contains":"mp3"}}},"sorting":{"order_by":"recorded_date","direction":"desc"},"paging":{"items":5},"projection":{"include":["id","recorded_date","site_id","original_file_name"]}}
+        '@
+      POWERSHELL
+    end
+
+    it 'accepts a standard filter object from the body of a POST' do
+      body = {
+        'filter' => {
+          'id' => {
+            'gt' => 2
+          },
+          'duration_seconds' => {
+            'range' => {
+              'interval' => '[1800, 3600]'
+            }
+          }
+        },
+        'paging' => {
+          'items' => 10
+        }
+      }
+      post '/audio_recordings/downloader', params: body, **api_with_body_headers(reader_token)
+      expect_success
+
+      expect(response.body).to include <<~POWERSHELL
+        $filter = @'
+        {"filter":{"id":{"gt":2},"duration_seconds":{"range":{"interval":"[1800, 3600]"}}},"sorting":{"order_by":"recorded_date","direction":"desc"},"paging":{"items":10},"projection":{"include":["id","recorded_date","site_id","original_file_name"]}}
+        '@
+      POWERSHELL
+    end
+
+    it 'send the script as an attachment with a file name' do
+      get '/audio_recordings/downloader', headers: api_request_headers(no_token)
+      expect_success
+      expect(response.headers['Content-Disposition'])
+        .to match(/attachment; filename="download_audio_files.ps1".*/)
+    end
+  end
+
+  describe 'the script successfully downloads files', :slow, timeout: 60 do
```

Comment:
I like the usage of `:slow` here

Reply:
Are these tests only run on linux when the CI is run? If so, its unlikely that these will be routinely tested on windows because they are slow tests

Reply:
Yeah there's only so much testing we can do in the dev environment. 

baw-server can not run on Windows at all, so to test this script we'd need to set up some kind of blackbox testing environment.

We'll see how it goes but I think (since we haven't seen it fail on Windows yet, and it is an OS agnostic script) we can revisit once this has seen some production time